src/cargo/core/compiler/mod.rs

```diff
@@ -1135,10 +1176,43 @@ fn on_stderr_line(
         struct CompilerMessage {
             rendered: String,
         }
-        if let Ok(error) = serde_json::from_str::<CompilerMessage>(compiler_message.get()) {
-            state.stderr(&error.rendered);
+        if let Ok(mut error) = serde_json::from_str::<CompilerMessage>(compiler_message.get()) {
+            // state.stderr will add a newline
+            if error.rendered.ends_with('\n') {
+                error.rendered.pop();
+            }
+            let rendered = if color {
+                error.rendered
+            } else {
+                strip_ansi_escapes::strip(&error.rendered)
+                    .map(|v| String::from_utf8(v).expect("utf8"))
+                    .unwrap_or(error.rendered)
```

Comment:
So taking a look at the source of this crate, I think it may be a bug that the `strip` function returns `io::Result` since it looks like the error only comes from the underlying `Write`, which in the case of `strip` is `Vec<u8>` which is infallible. In that sense it may be more clear to `.unwrap()` the result of `strip`?

(it felt a bit weird to not handle the utf-8 error but handle the `strip` error here)